metricflow-semantics/metricflow_semantics/naming/object_builder_scheme.py

```diff
@@ -80,14 +81,18 @@ def spec_pattern(self, input_str: str, semantic_manifest_lookup: SemanticManifes
                 ParameterSetField.DATE_PART,
             ]
 
+            time_granularity = None
```

Comment:
surprised this didn't require a type annotation 🤔

Reply:
Good call, I'll add one just to make future errors less confusing.

As for why it's not required, MyPy works in mysterious ways...... but in this case the type assignment flow is simple. In the end it can only be `None | ExpandedTimeGranularity` because there's just that one conditional branch below. If someone inserts another branch with a new potential type assignment that'll alter the type to `None | ExpandedTimeGranularity | whatever` and we'll get a confusing type error at the point where we try to pass this variable into `EntityLinkPatternParameterSet.from_parameters`.